Action step header legacy styling

mendelB opened the pull request:
### What does this PR do?
Adds ability to render Action Step Headers with legacy styling.


### Any background context you want to provide?
*My first pull request!!*
Basing a lot of this off of the logic @weerd implemented for the [`LedeBanner`](https://github.com/DoSomething/phoenix-next/blob/dev/resources/assets/components/LedeBanner/LedeBanner.js). Using a templates folder and `switch` logic checking which `campaign.template` is being rendered.


### What are the relevant tickets/cards?
https://www.pivotaltracker.com/story/show/150968514

### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.

deadlybutter commented on resources/assets/components/ActionPage/actionPage.scss:
```diff
@@ -62,6 +62,18 @@
     }
   }
 
+  .heading.-banner.legacy-step-header {
```
Hm let's remove the extra linebreak here. For the selector I think we can just make this `.legacy-step-header span`. As for the styling,

- replace `proxima-nova` with [this variable]($font-proxima-nova: "Proxima Nova", "Helvetica Neue", "Helvetica", "Arial", sans-serif;)
- if you need that specific padding format, use `$base-spacing` instead of 24px. However Im not sure you need this rule, is this overriding something or can you just use `padding-top`? Same for margin left, is that overriding something?

mendelB replied:
The `padding`, `margin-left`, and `width` are overriding some styling set for `.heading.-banner span`s (I think from forge)

deadlybutter replied:
Gotcha, ok! Also just confirmed with Luke in Slack, let's not use Proxima Nova here. You can use `$primary-font-family` which comes from the [Phoenix next toolbox](https://github.com/DoSomething/phoenix-next/blob/dev/resources/assets/scss/next-toolbox.scss). Feel free to ping Luke or Ashley in the future if you have a design or product question!

deadlybutter replied:
Also even though it's an override I would still replace the `24px` with `$base-spacing`

deadlybutter commented on resources/assets/components/ActionPage/templates/LegacyStepHeaderTemplate.js:
```diff
@@ -0,0 +1,20 @@
+  <FlexCell width="full">
+    <h2 className="heading -banner legacy-step-header">
+      <span>
+        Step { String(step) }: {title}
```
Hm, not crucial but this would probably read better a template literal

deadlybutter replied:
Also action steps don't always have a step number, there is a Contentful setting to disable that, so make sure it's not a required field and youre handling that if it's null. Checkout the Mosaic version which does this. Also did the mock want the actual step number or did it want the word version like the Mosaic? I don't remember from the mock

mendelB replied:
I know that the Drupal site shows the actual number, not sure about the mock though

deadlybutter commented on resources/assets/components/ActionPage/templates/MosaicStepHeaderTemplate.js:
```diff
@@ -0,0 +1,28 @@
+
+MosaicStepHeaderTemplate.propTypes = {
+  title: PropTypes.string.isRequired,
+  step: PropTypes.number.isRequired,
```
Again i wouldn't explicitly require step number here either, also `hideStepNumber` shouldnt be required, and make the default prop false.

deadlybutter commented:
I also just took a peak at the build failures. I would grab a plugin for your IDE that supports editor config so you're not worrying about the spacing & indents!

mendelB commented:
Thanks, @deadlybutter for the feedback! changed whatever I could based on your suggestions (save for the styling overriding the forge stuff), also added the editorConfig plugin to sublime, so hopefully, that'll prevent the styling snafus.

deadlybutter commented:
I think besides those quick style changes everything is good to go, but let's make sure @weerd gets a chance to weigh in as well

weerd commented:
> also added the editorConfig plugin to sublime, so hopefully, that'll prevent the styling snafus.

Yeah, I found that sometimes if editorconfig is added later it doesn't fix the files already updated, but will moving forward!

mendelB commented:
> Yeah, I found that sometimes if editorconfig is added later it doesn't fix the files already updated, but will moving forward!

Ahaaa, that would explain why it was only affecting certain files for me which was confusing.

weerd commented on resources/assets/components/ActionPage/StepHeader.js:
```diff
@@ -1,24 +1,37 @@
-  </FlexCell>
-);
+const StepHeader = ({ title, step, background, hideStepNumber, template }) => {
+  switch (template) {
```
🎉

weerd commented on resources/assets/components/ActionPage/actionPage.scss:
```diff
@@ -62,6 +62,15 @@
 
+  .heading.-banner.legacy-step-header span {
+    font-family: $primary-font-family;
+    padding: $base-spacing 0px 0px;
```
no need for unit specification when `0` is used as the value. `0` is unit-less 💃

weerd commented on resources/assets/components/ActionPage/templates/LegacyStepHeaderTemplate.js:
```diff
@@ -0,0 +1,27 @@
+
+const LegacyStepHeaderTemplate = ({ title, step, hideStepNumber }) => (
+  <FlexCell width="full">
+    <h2 className="heading -banner legacy-step-header">
```
we've deprecated the use of `-banner`. Use `-emphasized` instead! http://forge.dosomething.org/#heading :)

mendelB replied:
Oh nice! That seems to eliminate the need for any of the custom CSS on the `span`, save for the fact that the underline recedes due to a `width` reduction:
![image](https://user-images.githubusercontent.com/12417657/30488343-0b7dab9e-9a04-11e7-8526-4e691db954c4.png)
adding `width: 100%;` to the `h3` brings it back though, and I could use a better selector as per your comment below?
```
.legacy-step-header {
    width: 100%;
  }
```
resulting in:
![image](https://user-images.githubusercontent.com/12417657/30489126-6571371a-9a04-11e7-89ba-5cd2be8b2332.png)

weerd replied:
yeah, much preferred!

weerd commented on resources/assets/components/ActionPage/actionPage.scss:
```diff
@@ -62,6 +62,15 @@
     }
   }
 
+  .heading.-banner.legacy-step-header span {
```
It's a better practice not to overly specify a selector in CSS, because if we later need to override this, it makes it a lot more difficult! I think you'd be fine with just `.legacy-step-header span` right?

deadlybutter replied:
Oh yeah I left that in one of the other comments as well

weerd replied:
Also, I believe you should be able to just add this styling to the `h2` instead of specifying the `span`. In the legacy Phoenix-Ashes site we wrapped the text in a `span` for safe measure since adding the `::after` pseudo element for the yellow bar 😀

weerd commented:
@mendelB just confirming, does it look ok on smaller screens?

mendelB commented:
I toggled around on developer console through the various breakpoints and it seems to play nicely with all of em.

mendelB commented:
Thanks @weerd @deadlybutter for all the help and feedback!!! 

Am I meant to do the merge?

weerd commented:
Go for it @mendelB!
